File: prog10.c
#include<stdio.h>

int power();
int digit;

int main()
{
    printf("Enter the number of digits of the number: ");
    scanf("%i",&digit);

    int x;
    printf("Enter the number: ");
    scanf("%i",&x);

    int num;
    for (int i=1;x>0;i*=10)
    {
        int r=x%10;
        x=x/10;
        num+=r*(power()/i);
    }
    printf("Reversed Number: %i",num);
    
    return 0;
}

int power()
{
    int power=1;

    for (int j=1; j<digit; j++)
    {
        power*=10;
    }
    return power;
}

File: prog10.py
num=int(input("Enter the number: "))

num_list=[]
i=0
j=1
l=0
new_num=0

while num>0:
    rem=num%10
    num_list.append(rem)
    num=num//10
    i+=1

k=len(num_list)

while j<=len(num_list):
    new_num+=(10**(k-1))*(num_list[l])
    k-=1
    j+=1
    l+=1

print(f'Reversed number: {new_num}')